src/analysis/functions.rs

```diff
@@ -206,14 +210,32 @@ fn analyze_function(
     configured_functions: &[&config::functions::Function],
     imports: &mut Imports,
 ) -> Info {
-    let async = func.parameters.iter().any(|parameter| parameter.scope == ParameterScope::Async);
+    let async = func.name.ends_with("_async") ||
+                func.parameters.iter()
+                               .any(|parameter| parameter.scope == ParameterScope::Async &&
+                                                parameter.c_type == "GAsyncReadyCallback");
+    let expecting_data = !async &&
+                         func.parameters.iter()
+                                        .any(|par| par.c_type.ends_with("Func") ||
+                                                   par.c_type.ends_with("Callback"));
+
     let mut commented = false;
     let mut bounds: Bounds = Default::default();
     let mut to_glib_extras = HashMap::<usize, String>::new();
     let mut used_types: Vec<String> = Vec::with_capacity(4);
     let mut trampoline = None;
+    let mut callbacks = Vec::new();
+    let mut destroys = Vec::new();
     let mut async_future = None;
 
+    if !async && !expecting_data &&
+       func.parameters.iter().any(|par| par.c_type == "GDestroyNotify") {
+        warn!("Function \"{}\" with destroy callback without callbacks", func.name);
+        commented = true;
+    } else if async && !func.parameters.iter().any(|par| par.c_type == "GAsyncReadyCallback") {
```

Comment:
This case is impossible. You only set `async=true` above if such a parameter (and more things) were found

Reply:
Indeed, good catch!